### PR TITLE
Support `axis` parameter in DataFrame.drop

### PR DESCRIPTION
**Pull request by milesgranger**

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -878,12 +878,23 @@ def explode(self, column):
...
-    def drop(self, labels=None, columns=None, errors="raise"):
-        if columns is None:
-            columns = labels
-        if columns is None:
+    def drop(self, labels=None, axis=0, columns=None, errors="raise"):
+        if columns is None and labels is None:
             raise TypeError("must either specify 'columns' or 'labels'")
-        return new_collection(expr.Drop(self.expr, columns=columns, errors=errors))
+
+        if isinstance(axis, str):
```
Comment (phofl) [inline dask_expr/_collection.py:885]: We have a utility for this

Comment (milesgranger) [reply dask_expr/_collection.py:885]: Can you point me to it? I looked in `_util.py` w/o anything standing out.

Comment (phofl) [reply dask_expr/_collection.py:885]: Hm yikes it's a method in dask/dask that's defined on the DataFrame class

Please create a utility then, we will need this more often

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -1815,14 +1815,27 @@ def _simplify_up(self, parent):
             )
 
 
+def _drop_by_shallow_copy(df, labels=None, axis=0, columns=None, errors="raise"):
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply dask_expr/_expr.py]: Sure thing, https://github.com/dask-contrib/dask-expr/pull/442/commits/67a0d9bb2d7b76cb8222061ee82fb513d4b2e5f0

**Comment by phofl** [comment]
thx

**Pull request merged**